### ISSUE TITLE
Forcing API should accept mode numbers instead of physical wavenumbers

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#97]
## Problem

The current forcing API (`force_alfven_modes`, `force_slow_modes`) accepts physical wavenumbers `k_min` and `k_max`:

```python
def force_alfven_modes(
    state: KRMHDState,
    amplitude: float,
    k_min: float,  # Physical wavenumber [rad/L]
    k_max: float,  # Physical wavenumber [rad/L]
    dt: float,
    key: Array,
) -> Tuple[KRMHDState, Array]:
```

For a periodic domain with size L, the available Fourier modes are:
```
k = 2πn / L
```
where n = 1, 2, 3, ... are the mode numbers.

**This creates confusion** because users typically think in terms of mode numbers (n), not physical wavenumbers (k).

## Bug We Encountered

In `alfvenic_cascade_benchmark.py`, we attempted to force modes 1-3:

```python
Lx = Ly = Lz = 1.0  # Unit domain
k_force_min = 1.0   # Intended: mode n=1
k_force_max = 3.0   # Intended: mode n=3
```

But the actual fundamental mode is:
```
k₁ = 2π/L = 2π ≈ 6.28
```

**Result**: The forcing mask `(k_mag >= k_min) & (k_mag <= k_max)` was **empty** — no modes satisfied k ≤ 3.0 when the smallest available mode is k ≈ 6.28.

This caused all our turbulence simulations to run **without forcing** (energy injection rate ε_inj = 0), showing only decaying turbulence from initial conditions.

## Proposed Solution

Add optional parameters to accept mode numbers directly:

```python
def force_alfven_modes(
    state: KRMHDState,
    amplitude: float,
    k_min: float | None = None,  # Physical wavenumber (legacy)
    k_max: float | None = None,  # Physical wavenumber (legacy)
    n_min: int | None = None,    # Mode number (new)
    n_max: int | None = None,    # Mode number (new)
    dt: float,
    key: Array,
) -> Tuple[KRMHDState, Array]:
    """
    Force Alfvén modes in wavenumber band.
    
    Specify forcing range using either:
    - k_min, k_max: Physical wavenumbers [rad/L]
    - n_min, n_max: Mode numbers (converted to k = 2πn/L)
    """
    if n_min is not None and n_max is not None:
        # Convert mode numbers to physical wavenumbers
        Lx = state.grid.Lx
        k_min = n_min * 2 * jnp.pi / Lx
        k_max = n_max * 2 * jnp.pi / Lx
    elif k_min is None or k_max is None:
        raise ValueError("Must specify either (k_min, k_max) or (n_min, n_max)")
    
    # Rest of function unchanged...
```

## Benefits

1. **Intuitive**: Users think "force modes 1-3" not "force k ∈ [6.28, 18.85]"
2. **Less error-prone**: Eliminates 2π factor confusion
3. **Backward compatible**: Keep k_min/k_max for existing code
4. **Self-documenting**: `n_min=1, n_max=3` is clearer than `k_min=6.28, k_max=18.85`

## Workaround (Current)

Until this is implemented, users must manually convert mode numbers:

```python
n_force_min = 1
n_force_max = 3
k_force_min = n_force_min * 2 * np.pi / Lx
k_force_max = n_force_max * 2 * np.pi / Lx

state, key = force_alfven_modes(state, amplitude, k_force_min, k_force_max, dt, key)
```

## Related Files

- `src/krmhd/forcing.py`: `force_alfven_modes()`, `force_slow_modes()`
- `src/krmhd/forcing.py`: `gaussian_white_noise_fourier()` (internal implementation)
- `examples/alfvenic_cascade_benchmark.py`: Example that hit this bug
